syntax = "proto3";

package metadata_service.v1;

import "google/api/annotations.proto";
import "messages/image_dto/v1/image_dto.proto";

// Request to get caption for an image
message UpdateIPTCDataRequest {
	// Image which 
	shared.messages.image_dto.v1.ImageDto image = 1;
	string caption = 2;
	string hashtag_csv = 3;
}

message UpdateIPTCDataResponse {
}

// Service handling all image metadata related operations
service MetaDataService {
	// Update the IPTC data for an image with the given caption and hashtags
	rpc update_iptc_data(UpdateIPTCDataRequest) returns (UpdateIPTCDataResponse) {
		option (google.api.http) = {
			post: "/v1/iptc"
			body: "*"
		};
	}
}
